### PR TITLE
Drop support for "use math"

### PR DESCRIPTION
**Pull request by saghul**

### CODE REVIEW

**Comment by saghul** [comment]
Oh, I broke something and 262 found it! I'll take a look later.

**Comment by saghul** [comment]
This one is ready @bnoordhuis !

File: quickjs.c [hunk 1]
```diff
@@ -10086,9 +10076,7 @@ static double js_strtod(const char *p, int radix, BOOL is_float)
 #define ATOD_TYPE_BIG_INT     (1 << 7)
 #define ATOD_TYPE_BIG_FLOAT   (2 << 7)
 #define ATOD_TYPE_BIG_DECIMAL (3 << 7)
-/* assume bigint mode: floats are parsed as integers if no decimal
-   point nor exponent */
-#define ATOD_MODE_BIGINT      (1 << 9) 
+#define ATOD_MODE_UNUSED      (1 << 9)
```
Comment (bnoordhuis) [inline quickjs.c:10079]:
```suggestion
```
Or did you have a reason for keeping it around?

File: quickjs.c [hunk 2]
```diff
@@ -10324,29 +10312,11 @@ static JSValue js_atof(JSContext *ctx, const char *str, const char **pp,
...
-                    goto fail;
-            } else {
-                if (is_float && radix != 10)
-                    goto fail;
-            }
+        } else if (is_float && radix != 10) {
+            goto fail;
         }
+    } else if ((atod_type == ATOD_TYPE_FLOAT64) && is_float && radix != 10) {
+            goto fail;
```
Comment (bnoordhuis) [inline quickjs.c]:
```suggestion
    } else if (atod_type == ATOD_TYPE_FLOAT64 && is_float && radix != 10) {
        goto fail;
```

File: quickjs.c [hunk 3]
```diff
@@ -12163,43 +12123,19 @@ static bf_t *JS_ToBigIntFree(JSContext *ctx, bf_t *buf, JSValue val)
     case JS_TAG_INT:
     case JS_TAG_NULL:
     case JS_TAG_UNDEFINED:
-        if (!is_math_mode(ctx))
-            goto fail;
+    case JS_TAG_FLOAT64:
+    case JS_TAG_BIG_FLOAT:
+        goto fail;
         /* fall tru */
```
Comment (bnoordhuis) [inline quickjs.c:12129]:
```suggestion
```

File: quickjs.c [hunk 4]
```diff
@@ -21082,10 +20783,7 @@ static __exception int next_token(JSParseState *s)
     case '^':
         if (p[1] == '=') {
             p += 2;
-            if (s->cur_func->js_mode & JS_MODE_MATH)
-                s->token.val = TOK_MATH_POW_ASSIGN;
-            else
-                s->token.val = TOK_XOR_ASSIGN;
+            s->token.val = TOK_XOR_ASSIGN;
```
Comment (bnoordhuis) [inline quickjs.c]: I believe you can replace this block (the code inside the `#ifdef CONFIG_BIGNUM`) with just the `#else` case.

**Review by bnoordhuis — APPROVED** [review]
:+1: 

I did have a ready-to-submit commit that removes all the `#ifdef CONFIG_BIGNUM` that's probably going to conflict heavily with this. Oh well.

**Comment by saghul** [comment]
> 👍
> 
> I did have a ready-to-submit commit that removes all the `#ifdef CONFIG_BIGNUM` that's probably going to conflict heavily with this. Oh well.

Sorry I got the impression you hadn't started :-/

**Comment by bnoordhuis** [comment]
Don't worry about it, it happens.

File: quickjs.c (hunk 1 above)
Comment (saghul) [reply quickjs.c:10079]: Nah.

**Pull request merged**